platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/tipset/ChildlessEventTrackerTests.java

```diff
@@ -1,158 +1,325 @@
+    void testHighestGenBranchedEventsAreTracked() {
         final Random random = getRandomPrintSeed();
-
         final ChildlessEventTracker tracker = new ChildlessEventTracker();
+        final NodeId nodeId = NodeId.of(0);
 
-        final EventDescriptorWrapper e0 = newEventDescriptor(randomHash(random), NodeId.of(0), 0);
-        final EventDescriptorWrapper e1 = newEventDescriptor(randomHash(random), NodeId.of(0), 1);
-        final EventDescriptorWrapper e2 = newEventDescriptor(randomHash(random), NodeId.of(0), 2);
+        final PlatformEvent e0 =
+                new TestingEventBuilder(random).setCreatorId(nodeId).setNGen(0).build();
+        final PlatformEvent e1 =
+                new TestingEventBuilder(random).setCreatorId(nodeId).setNGen(1).build();
+        final PlatformEvent e2 =
+                new TestingEventBuilder(random).setCreatorId(nodeId).setNGen(2).build();
 
-        tracker.addEvent(e0, List.of());
-        tracker.addEvent(e1, List.of(e0));
-        tracker.addEvent(e2, List.of(e1));
+        tracker.addEvent(e0);
+        tracker.addEvent(e1);
+        tracker.addEvent(e2);
 
-        final List<EventDescriptorWrapper> batch1 = tracker.getChildlessEvents();
-        assertEquals(1, batch1.size());
-        assertEquals(e2, batch1.get(0));
+        assertThat(tracker.getChildlessEvents().size()).isEqualTo(1);
+        assertThat(tracker.getChildlessEvents().getFirst()).isEqualTo(e2);
 
-        final EventDescriptorWrapper e3 = newEventDescriptor(randomHash(random), NodeId.of(0), 3);
-        final EventDescriptorWrapper e3Branch = newEventDescriptor(randomHash(random), NodeId.of(0), 3);
+        final PlatformEvent e3 = new TestingEventBuilder(random)
+                .setCreatorId(nodeId)
+                .setSelfParent(e2)
+                .setNGen(3)
+                .build();
+        final PlatformEvent e3Branch = new TestingEventBuilder(random)
+                .setCreatorId(nodeId)
+                .setSelfParent(e2)
+                .setNGen(3)
+                .build();
 
         // Branch with the same generation, existing event should not be discarded.
-        tracker.addEvent(e3, List.of(e2));
-        tracker.addEvent(e3Branch, List.of(e2));
+        tracker.addEvent(e3);
+        tracker.addEvent(e3Branch);
 
-        final List<EventDescriptorWrapper> batch2 = tracker.getChildlessEvents();
-        assertEquals(1, batch2.size());
-        assertEquals(e3, batch2.get(0));
+        assertThat(tracker.getChildlessEvents().size()).isEqualTo(1);
+        assertThat(tracker.getChildlessEvents().getFirst()).isEqualTo(e3);
 
         // Branch with a lower generation, existing event should not be discarded.
-        final EventDescriptorWrapper e2Branch = newEventDescriptor(randomHash(random), NodeId.of(0), 2);
-        tracker.addEvent(e2Branch, List.of(e1));
+        final PlatformEvent e2Branch = new TestingEventBuilder(random)
```

Comment:
I'd prefer to put this in a separate test. Many small tests are easier for me to digest. 🙂